editor/scene/3d/gizmos/physics/soft_body_3d_gizmo_plugin.cpp

```diff
@@ -106,7 +107,13 @@ Variant SoftBody3DGizmoPlugin::get_handle_value(const EditorNode3DGizmo *p_gizmo
 
 void SoftBody3DGizmoPlugin::commit_handle(const EditorNode3DGizmo *p_gizmo, int p_id, bool p_secondary, const Variant &p_restore, bool p_cancel) {
 	SoftBody3D *soft_body = Object::cast_to<SoftBody3D>(p_gizmo->get_node_3d());
-	soft_body->pin_point_toggle(p_id);
+	const bool is_pinned = soft_body->is_point_pinned(p_id);
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(vformat(TTR("%s SoftBody3D pinned point %d."), is_pinned ? "Remove" : "Add", p_id));
```

Comment:
Don't put Add/Remove as separate strings, because they won't be translated.

Reply:
I've added the suggestion in https://github.com/godotengine/godot/pull/109828 (same for the other one below).